gradle-guide-error-prone/src/test/java/com/palantir/gradle/guide/errorprone/taskexecution/GetProjectCopyTest.java

```diff
@@ -0,0 +1,176 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.guide.errorprone.taskexecution;
+
+import org.junit.jupiter.api.Test;
+
+@SuppressWarnings("LineLength")
+public class GetProjectCopyTest extends IllegalMethodCalledDuringTaskExecutionTest {
+    @Test
+    void projectLayout_not_injected_should_fix_and_inject() {
+        testFix(
+                "AlreadyAbstractTask.java",
+                """
+                import java.io.File;
+                import org.gradle.api.DefaultTask;
+                import org.gradle.api.Plugin;
+                import org.gradle.api.Project;
+                import org.gradle.api.tasks.TaskAction;
+                abstract class AlreadyAbstractTask extends DefaultTask {
+                    @TaskAction
+                    final void action() {
+                        getProject().copy(copySpec -> {
+                            copySpec.from("src");
+                            copySpec.into("dest");
+                        });
+                    }
+                }
+                """,
+                """
+                import java.io.File;
+                import javax.inject.Inject;
+                import org.gradle.api.DefaultTask;
+                import org.gradle.api.Plugin;
+                import org.gradle.api.Project;
+                import org.gradle.api.file.FileSystemOperations;
+                import org.gradle.api.tasks.TaskAction;
+                abstract class AlreadyAbstractTask extends DefaultTask {
+                    @Inject
+                    protected abstract FileSystemOperations getFileSystemOperations();
+                    @TaskAction
+                    final void action() {
+                        getFileSystemOperations().copy(copySpec -> {
+                            copySpec.from("src");
+                            copySpec.into("dest");
+                        });
+                    }
+                }
+                """);
+    }
+
+    @Test
+    void fileSystemOperations_already_injected_should_fix_without_injecting_again() {
+        testFix(
+                "AlreadyInjectedTask.java",
+                """
+                import java.io.File;
+                import javax.inject.Inject;
+                import org.gradle.api.DefaultTask;
+                import org.gradle.api.Plugin;
+                import org.gradle.api.Project;
+                import org.gradle.api.file.FileSystemOperations;
+                import org.gradle.api.tasks.TaskAction;
+                abstract class AlreadyInjectedTask extends DefaultTask {
+                    private static final String IM_AT_THE_TOP_OF_THE_CLASS = "happy_squirrel.txt";
+                    @Inject
+                    protected abstract FileSystemOperations getFileSystemOperations();
```

Comment:
I'd be curious to see if you had a test where you already had a `FileSystemOperations` but named something else.

Reply:
It's not great but I think you can just inject the same service multiple times